### PR TITLE
Add output_transform type check in Metric class (#3352)

### PR DESCRIPTION
**Pull request by HyeSungP**
Implement #3352

Check list:

- [x] Add above code
- [x] Add a test checking that error is raised to

### CODE REVIEW

File: ignite/metrics/metric.py [hunk 1]
```diff
@@ -361,6 +361,8 @@ def __init__(
         device: Union[str, torch.device] = torch.device("cpu"),
         skip_unrolling: bool = False,
     ):
+        if not callable(output_transform):
+            raise TypeError("Argument output_transform should be callable, " f"got {type(output_transform)}")
```
Comment (vfdev-5) [inline ignite/metrics/metric.py:365]: Please fix the f-string here

File: tests/ignite/metrics/test_metric.py [hunk 2]
```diff
@@ -1468,3 +1468,19 @@ def test_access_to_metric_dunder_attributes():
...
+    def update(self):
+        pass
+
+    def compute(self):
+        pass
+
+
+def test_output_transform_type_check():
+    with pytest.raises(TypeError, match="Argument output_transform should be callable"):
+        DummyMetric7(output_transform=1)
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_metric.py:1486]: Let's remove the definition DummyMetric7 and DummyMetric1 for the test

File: ignite/metrics/metric.py (hunk 1 above)
Comment (HyeSungP) [reply ignite/metrics/metric.py]: Thanks! I've updated the f-string formatting accordingly.

File: tests/ignite/metrics/test_metric.py (hunk 2 above)
Comment (HyeSungP) [reply tests/ignite/metrics/test_metric.py:1486]: @vfdev-5 Thanks for the advice. Would it be better to define a single, shared DummyMetric class for these tests, or should we define the class within each test function, as is done in some other tests?

Comment (vfdev-5) [reply tests/ignite/metrics/test_metric.py:1486]: Thanks for pointing out this! Maybe, later we can improve this part. I do not think we can define a single shared DummyMetric for all test cases. Probably, there are also some tests using some globally defined DummyMetric, so both cases can exist.
For our purposes I think we can reuse DummyMetric1

Comment (vfdev-5) [reply tests/ignite/metrics/test_metric.py:1486]: @HyeSungP can you please update the test according to the last comment: https://github.com/pytorch/ignite/pull/3371#discussion_r2028357790

-> For our purposes I think we can reuse DummyMetric1 and remove DummyMetric7

Comment (HyeSungP) [reply tests/ignite/metrics/test_metric.py]: @vfdev-5 Sorry for the delay. I've updated the code based on your advice. Could you please review it?

File: tests/ignite/metrics/test_metric.py [hunk 3]
```diff
@@ -1468,3 +1468,10 @@ def test_access_to_metric_dunder_attributes():
 
     # `inspect.signature` accesses `__signature__` attribute of the metric.
     assert "value" in inspect.signature(metric).parameters.keys()
+
+
+def test_output_transform_type_check():
+    y_pred = torch.tensor([[2.0], [-2.0]])
+    y = torch.zeros(2)
+    with pytest.raises(TypeError, match="Argument output_transform should be callable"):
+        DummyMetric1(true_output=(y_pred, y), output_transform=1)
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_metric.py:1477]: You can probably set treu output to none. But ok to have it this way.

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @HyeSungP !
